main.go

```diff
@@ -123,22 +86,50 @@ func runTemplate(c *cli.Context) error {
 
 	g := NewGomplate(data, lDelim, rDelim)
 
-	inputs := readInputs(c.String("in"), c.StringSlice("file"))
+	if err := validateInOutOptions(c); err != nil {
+		return err
+	}
 
-	outputs := c.StringSlice("out")
-	if len(outputs) == 0 {
-		outputs = []string{"-"}
+	inputDir := c.String("input-dir")
+	if inputDir != "" {
+		return processInputDir(inputDir, getOutputDir(c), g)
 	}
 
-	for n, input := range inputs {
-		out, err := openOutFile(outputs[n])
-		if err != nil {
-			return err
-		}
-		defer out.Close() // nolint: errcheck
-		g.RunTemplate(input, out)
+	return processInputFiles(c.String("in"), c.StringSlice("file"), c.StringSlice("out"), g)
+}
+func getOutputDir(c *cli.Context) string {
+	out := c.String("output-dir")
+	if out != "" {
+		return out
 	}
+	return "."
```

Comment:
I tried that, but this doesn't work in the validation logic, which checks for whether the outputDir is set. if it has a default value, then it is always set. You then also can't distinguish between people setting "." explicitely (and so switching on the ferature) or whether it comes from the default.

Dont think its so bad here.